src/Compilers/CSharp/Test/Semantic/Semantics/RawInterpolationTests_Handler.cs

```diff
@@ -4771,25 +4771,44 @@ .locals init (CustomHandler V_0)
 ");
     }
 
-    [Theory]
-    [InlineData(@"$""""""{1,2:f}Literal""""""")]
-    [InlineData(@"$""""""{1,2:f}"""""" + $""""""Literal""""""")]
-    public void PassAsRefWithoutKeyword_02(string expression)
+    [Fact]
+    public void PassAsRefWithoutKeyword_02_ConstantInterpolatedString()
     {
-        var code = @"
-M(" + expression + @");
-M(ref " + expression + @");
+        var code = """"
+                M($"""{1,2:f}Literal""");
+                M(ref $"""{1,2:f}Literal""");
 
-void M(ref CustomHandler c) => System.Console.WriteLine(c);";
+                void M(ref CustomHandler c) => System.Console.WriteLine(c);
+                """";
 
         var comp = CreateCompilation(new[] { code, GetInterpolatedStringCustomHandlerType("CustomHandler", "class", useBoolReturns: false) }, targetFramework: TargetFramework.Net50);
         comp.VerifyDiagnostics(
-            // (2,3): error CS1620: Argument 1 must be passed with the 'ref' keyword
+            // (1,3): error CS1620: Argument 1 must be passed with the 'ref' keyword
             // M($"{1,2:f}Literal");
-            Diagnostic(ErrorCode.ERR_BadArgRef, expression).WithArguments("1", "ref").WithLocation(2, 3),
-            // (3,7): error CS1510: A ref or out value must be an assignable variable
+            Diagnostic(ErrorCode.ERR_BadArgRef, @"$""""""{1,2:f}Literal""""""").WithArguments("1", "ref").WithLocation(1, 3),
+            // (2,7): error CS1510: A ref or out value must be an assignable variable
+            // M(ref $"{1,2:f}Literal");
+            Diagnostic(ErrorCode.ERR_RefLvalueExpected, @"$""""""{1,2:f}Literal""""""").WithLocation(2, 7));
+    }
+
+    [Fact]
+    public void PassAsRefWithoutKeyword_02_SumOfInterpolatedStrings()
+    {
+        var code = """"
+                M($"""{1,2:f}""" + $"""Literal""");
+                M(ref $"""{1,2:f}""" + $"""Literal""");
+
+                void M(ref CustomHandler c) => System.Console.WriteLine(c);
+                """";
+
+        var comp = CreateCompilation(new[] { code, GetInterpolatedStringCustomHandlerType("CustomHandler", "class", useBoolReturns: false) }, targetFramework: TargetFramework.Net50);
+        comp.VerifyDiagnostics(
+            // (1,3): error CS9203: Expected interpolated string
```

Comment:
Sorry, the second comment I made here was wrong.

> At the same time you said below that it would be nice to treat "a" + "b" as a single string,

Right. That's not-interpolated-string + not-interpolated-string. The error I commented on here in commit 3 was on interpolated-string + interpolated-string: that can be converted to a handler just fine, provided that everything else binds correctly, so an "expected interpolated string" comment there is wrong.